ports/esp32/network_common.c

```diff
@@ -154,6 +155,146 @@ static mp_obj_t esp_ifconfig(size_t n_args, const mp_obj_t *args) {
+                        if (self->if_id == ESP_IF_WIFI_STA || self->if_id == ESP_IF_ETH) {
+                            esp_exceptions(esp_netif_dhcpc_get_status(self->netif, &status));
+                            if (mp_obj_is_true(e->value) && status != ESP_NETIF_DHCP_STARTED) {
+                                esp_exceptions(esp_netif_dhcpc_start(self->netif));
+                            } else if (!mp_obj_is_true(e->value) && status == ESP_NETIF_DHCP_STARTED) {
+                                esp_exceptions(esp_netif_dhcpc_stop(self->netif));
+                            }
+                        } else {
+                            mp_raise_ValueError(MP_ERROR_TEXT("unexpected key"));
+                            break;
+                        }
+                        break;
+                    }
+                    case MP_QSTR_addr4: {
+                        int prefix_bits = 32;
+                        if (e->value != mp_const_none && mp_obj_is_str(e->value)) {
+                            size_t addr_len;
+                            const char *input_str = mp_obj_str_get_data(e->value, &addr_len);
+                            char plain_ip[IPADDR_STRLEN_MAX];
+                            char *split = strchr(input_str, '/');
+                            const char *addr_str = input_str;
+                            if (split) {
+                                int to_copy = sizeof(plain_ip) - 1;
+                                if (split - addr_str < to_copy) {
+                                    to_copy = split - addr_str;
+                                }
+                                memcpy(plain_ip, addr_str, to_copy);
+                                mp_obj_t prefix_obj = mp_parse_num_integer(split + 1, strlen(split + 1), 10, NULL);
+                                prefix_bits = mp_obj_get_int(prefix_obj);
+                            }
+                            uint32_t mask = -(1u << (32 - prefix_bits));
+                            uint32_t *m = (uint32_t *)&info.netmask;
+                            *m = esp_netif_htonl(mask);
+                            netutils_parse_ipv4_addr(e->value, (void *)&info.ip, NETUTILS_BIG);
+                        } else if (e->value != mp_const_none) {
+                            mp_obj_t *items;
+                            mp_obj_get_array_fixed_n(e->value, 2, &items);
+                            netutils_parse_ipv4_addr(items[0], (void *)&info.ip, NETUTILS_BIG);
+                            netutils_parse_ipv4_addr(items[1], (void *)&info.netmask, NETUTILS_BIG);
+                        }
+                        touched_ip_info = 1;
+                        break;
+                    }
+                    case MP_QSTR_gw4: {
+                        netutils_parse_ipv4_addr(e->value, (void *)&info.gw, NETUTILS_BIG);
+                        touched_ip_info = 1;
+                        break;
+                    }
+                    case MP_QSTR_dns: {
```

Comment:
For esp, i've moved `dns` to the interface-local variant, it seems dns config is set per-interface here.

Reply:
At least that's less code.

Reply:
You have dns setting in network.ipconfig. Why do you keep it here at nic.ipconfig (and removed dns query here)?

Reply:
ooh, yes. I missed that. Does the `network.ipconfig(dns=)` work? Then I'll remove the remainder here.

Reply:
`network.ipconfig(dns=)` works.